### DISCUSSION TITLE
Inventory / Sales management system

### DISCUSSION THREAD

**Discussion by yewkay** [Budibase/budibase#7337] [Help]
Hi I'm new to budibase and database apps on the whole. I like what I see so far and managed to follow through a few youtube tutorials. 

I am trying to create a sales order cum inventory management system. 

My sales order process has 4 main inputs
Product - Product we're selling 
Quantity - quantity of product to sell in this order
Status - 3 options: released, reserved and confirmed
Expiry date - the date to reserve the item. upon expiry date, if the sales order status is not confirmed, it will be set to release. If status is reserved or confirmed, the quantity of the above products will be deducted from the database. if it's set to release, whatever quantity reduced earlier will be released back into the database.

Where do i go to key in the logic to reduce the quantity of the product from the database when the status is reserved? and automatically set the status to released and returns the stock back into the database when it has passed the expiry date?

**Comment by melohagan** [comment]
Hey @yewkay 

> Where do i go to key in the logic to reduce the quantity of the product from the database when the status is reserved?
Use an automation with a **Row Updated** trigger. 
Then use <a href="https://docs.budibase.com/docs/conditions-1">conditions</a> to only reduce the quantity of the product if status is *reserved*.

> set the status to released and returns the stock back into the database when it has passed the expiry date
I think what you would look to do is use a Cron job automation: https://docs.budibase.com/docs/cronjob

That way you can <a href="https://docs.budibase.com/docs/automation-actions#looping">loop</a> over the rows and check if they're past their expiry date. 
Use a condition, as mentioned above, if true then set the status to *released*. 

To make the looping more efficient it may be wise to have separate tables for each status. That way with the automations they would act as a queue of sorts.
